subsys/dfu/src/dfu_target_modem.c

```diff
@@ -195,6 +195,14 @@ int dfu_target_modem_write(const void *const buf, size_t len)
 		} else {
 			return 0;
 		}
+	case DFU_AREA_NOT_BLANK:
+		delete_banked_modem_fw();
+		err = dfu_target_modem_write(buf, len);
+		if (err < 0) {
+			return -EINVAL;
+		} else {
+			return 0;
+		}
```

Comment:
Unrelated, but I recommed adding

    default:
        return -EFAULT;

for readability,

Reply:
Think I've added it the way you wanted now.